src/main/scala/tech/sourced/gemini/cmd/Parser.scala

```diff
@@ -0,0 +1,43 @@
+package tech.sourced.gemini.cmd
+
+abstract class Parser[C](programName: String) extends scopt.OptionParser[C](programName) {
+  /**
+    * parses the given `args` but reads environment variables also
+    */
+  def parseWithEnv(args: Seq[String], init: C): Option[C] = {
+    val shortToFullMapping = options
+      .filter(opt => opt.shortOpt.isDefined)
+      .foldLeft(Map[String, String]()) { case (opts, opt) =>
+        opts + ("-" + opt.shortOpt.get -> opt.fullName)
+      }
+
+    val cliNames = args
+      .filter(arg => arg.startsWith("-")) // remove positional arguments
+      .map(arg => arg.split("=", 2)(0))
+      .map(name => shortToFullMapping.getOrElse(name, name)) // convert short names to full names
+
+    val envNames = options
+      .map(opt => opt.fullName)
+      .filter(name => name.startsWith("--")) // remove positional arguments
+      .filterNot(name => name == "verbose") // we don't support verbose as env var
+      .filterNot(name => cliNames.contains(name)) // don't read env vars for passed flags
```

Comment:
Could you please help me understand high-level logic of ENV VAR vs CLI flags?

Usual expectations from configuration subsystem are:
 - all configuration options (except input path, which is always passed though positional arg) can be set either though `ENV_VAR` or though CLI flags (`--env-var`, `-e`)
 - one form has precedence over another i.e CLI flags always override ENV VARs

This has benefits of simplicity and predictability for the end user.

Would that be the case right now or do you think there is a better option? Thanks!

Reply:
yes. It works exactly as you described. I hoped comments in the code make it clear.

Reply:
Nice! Thank you, was confused with that some env vars are not read.

> we don't support verbose as env var

What is the reason for that?

Reply:
The code doesn't read env vars which were provided by flags already. (they will override env vars anyway but it would make code more complicated)

> we don't support verbose as env var. What is the reason for that?
currently, in master, we don't read verbose env variable. The PR is refactoring. Refactoring usually shouldn't change the way how the code works only internals.

though I don't see any problem with enabling it.

Reply:
Awesome, yes, thank you again for explanation!